### ISSUE TITLE
.half throws error during training when using batchnorm

### ISSUE THREAD

**Issue by ashtonsix** [fastai/fastai#507]
I've spent hours stuck on this!

![error: "expected HalfTensor, recieved FloatTensor"](https://user-images.githubusercontent.com/6740947/40580534-0d6e9886-6138-11e8-9830-fb6ea2e66d78.png)

Inside FP16 I tried replacing:

```
self.module = batchnorm_to_fp32(module.half())
```

with

```
self.module = module.half()
```

& it worked, why isn't this the default?

**Comment by jph00** [comment]
Because of this: https://discuss.pytorch.org/t/training-with-half-precision/11815

**Comment by ashtonsix** [comment]
Hmm... this lady says FP16 batch-norm only lead to a 0.5% decrease in accuracy, think I can stomach that for 10x faster training on tensor cores.

https://github.com/pytorch/pytorch/issues/520#issuecomment-277741834
